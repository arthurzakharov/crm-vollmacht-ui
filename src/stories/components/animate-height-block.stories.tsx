import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { AnimateHeight } from "../../components";
import MaxWidthDecorator from "../decorators/max-width-decorator";

type Story = StoryObj<typeof AnimateHeight>;

const meta: Meta<typeof AnimateHeight> = {
  title: "COMPONENTS/AnimateHeight",
  component: AnimateHeight,
  decorators: [MaxWidthDecorator],
  parameters: {
    MaxWidthDecorator: 768,
  },
  args: {
    closed: false,
    animateOpacity: true,
    duration: 1000,
    delay: 500,
    easing: "cubic-bezier(0.4, 0, 0.2, 1)",
  },
  render: (args) => (
    <AnimateHeight {...args}>
      <div
        style={{
          height: 300,
          backgroundImage:
            "linear-gradient(45deg, #d3d3d3 25%, transparent 25%, transparent 50%, #d3d3d3 50%, #d3d3d3 75%, transparent 75%, transparent 100%)",
          backgroundSize: "56.57px 56.57px",
        }}
      />
    </AnimateHeight>
  ),
};

export const Closed: Story = {
  name: "Closed",
  args: {
    closed: true,
  },
};

export const Opened: Story = {
  name: "Opened",
  args: {
    closed: false,
  },
};

export const DefaultValues: Story = {
  name: "DefaultValues",
  args: {
    animateOpacity: undefined,
    duration: undefined,
    delay: undefined,
    easing: undefined,
  },
};

export default meta;
