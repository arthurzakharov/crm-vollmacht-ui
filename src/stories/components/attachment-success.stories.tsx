import { Meta, StoryObj } from "@storybook/react";
import { AttachmentSuccess } from "../../components";
import MaxWidthDecorator from "../decorators/max-width-decorator";
import submittedSrc from "../assets/submitted.png";

type Story = StoryObj<typeof AttachmentSuccess>;

const meta: Meta<typeof AttachmentSuccess> = {
  title: "COMPONENTS/AttachmentSuccess",
  component: AttachmentSuccess,
  parameters: {
    MaxWidthDecorator: 600,
  },
  decorators: [MaxWidthDecorator],
};

export const Default: Story = {
  name: "Attachment Success",
  args: {
    icon: submittedSrc,
    title: "Vielen Dank!",
    text: [
      "Wir haben Ihre Informationen erhalten und werden diese umgehend bearbeiten.",
      "Sobald neue Erkentnisse vorliegen, setzen wir uns mit Ihnen in Verbindung.",
    ],
  },
};

export default meta;
