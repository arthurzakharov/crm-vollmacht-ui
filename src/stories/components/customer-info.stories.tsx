import { Meta, StoryObj } from "@storybook/react";
import { fn, within, expect } from "@storybook/test";
import { CustomerInfo } from "../../components";
import MaxWidthDecorator from "../decorators/max-width-decorator";
import { Delayed } from "../utils";

type Story = StoryObj<typeof CustomerInfo>;

const meta: Meta<typeof CustomerInfo> = {
  title: "COMPONENTS/CustomerInfo",
  component: CustomerInfo,
  parameters: {
    MaxWidthDecorator: 295,
  },
  decorators: [MaxWidthDecorator],
  args: {
    version: "form",
    list: ["Arthur Zakharov", "Geb in 12/12/2000", "Marienstr. 7, 01067 Dresden"],
    isEditButtonVisible: false,
    onClick: fn(),
  },
};

export const VersionForm: Story = {
  name: "Version Form",
  args: {
    version: "form",
  },
};

export const VersionSidebar: Story = {
  name: "Version Sidebar",
  args: {
    version: "sidebar",
  },
};

export const EmptyList: Story = {
  name: "Empty List",
  args: {
    list: [],
  },
};

export const EditButtonHidden: Story = {
  name: "Edit Button Hidden",
  args: {
    isEditButtonVisible: false,
  },
};

export const EditButtonVisible: Story = {
  name: "Edit Button Visible",
  args: {
    isEditButtonVisible: true,
  },
  play: async ({ args, canvasElement, step }) => {
    const button = within(canvasElement).getByRole("button");
    await step("Click edit button, focus not active", async () => {
      await Delayed.click(button);
      await expect(args.onClick).toHaveBeenNthCalledWith(1);
      await expect(button).not.toHaveFocus();
    });
    await step("Click with keyboard, focus not active", async () => {
      await Delayed.tab();
      await Delayed.keyboard("[Space]");
      await expect(args.onClick).toHaveBeenNthCalledWith(2);
      await expect(button).not.toHaveFocus();
    });
  },
};

export const DefaultValues: Story = {
  name: "Default Values",
  args: {
    isEditButtonVisible: undefined,
  },
};

export default meta;
