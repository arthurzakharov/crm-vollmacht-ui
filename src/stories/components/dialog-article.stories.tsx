import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { Dialog, DialogArticle } from "../../components";
import MaxWidthDecorator from "../decorators/max-width-decorator";
import PositionFixedDecorator from "../decorators/position-fixed-decorator";

type Story = StoryObj<typeof DialogArticle>;

const meta: Meta<typeof DialogArticle> = {
  title: "COMPONENTS/Dialog Article",
  component: DialogArticle,
  decorators: [MaxWidthDecorator, PositionFixedDecorator],
  parameters: {
    MaxWidthDecorator: 600,
  },
  args: {
    hasNoCloseButton: false,
    cancelButton: "",
    confirmButton: "",
    onCancel: fn(),
    onConfirm: fn(),
    onClose: fn(),
  },
  render: (args) => {
    return (
      <Dialog
        dialogs={[
          {
            name: "dialog",
            element: (
              <DialogArticle {...args}>
                <div>
                  <h1>Allgemeine Informationen bei Verträgen im elektronischen Rechtsverkehr</h1>
                  <p>Stand: 01.11.2023</p>
                  <p>
                    Allgemeine Informationen zu Verträgen, die über www.schuldenanalyse-kostenlos.de zwischen der
                    rightmart Berlin Rechtsanwaltsgesellschaft mbH, Sonnenallee 260/262, 12057 Berlin - im Folgenden
                    „Kanzlei“ - und dem Mandanten - im Folgenden „Mandant“ - geschlossen werden.
                  </p>
                  <h2>§ 1 Vertragsgegenstand und -inhalt, Zustandekommen des Vertrages</h2>
                  <p>
                    (1) Gegenstand des Vertrags ist die Beratung und Vertretung des Mandanten bei der außergerichtlichen
                    Schuldenregulierung durch Mitarbeiter der Kanzlei.
                  </p>
                  <p>
                    (2) Der Vertrag kommt zustande mit: <br />
                    <br />
                    rightmart Berlin Rechtsanwaltsgesellschaft mbH <br />
                    Sonnenallee 260/262 <br />
                    12057 Berlin
                  </p>
                  <p>
                    (3) Die Leistung der Kanzlei besteht in der Beratung und Vertretung des Mandanten bei der
                    außergerichtlichen Regulierung seiner Schulden. Der Mandant bestätigt durch Anklicken des Buttons
                    „Kostenpflichtig beauftragen“ seinen verbindlichen Auftrag an die Kanzlei. Mit Verbindlichkeit des
                    Auftrags verpflichtet sich der Mandant zur Begleichung der nach der Vergütungsvereinbarung
                    geschlossenen Vergütung. Sobald die Kanzlei den Auftrag des Mandanten erhalten hat, wird dem
                    Mandanten eine Bestätigung über den Eingang seines Auftrages bei der Kanzlei zugesandt, in der Regel
                    per E-Mail (Eingangsbestätigung).
                  </p>
                  <p>(4) Die Beauftragung über die Webseite der Kanzlei funktioniert wie folgt:</p>
                  <p>
                    <b>Angaben zur Person:</b> <br />
                    Hier trägt der Mandant seine Angaben zur Person einschließlich Geburtsdatum und Geburtsort in die
                    jeweils dafür vorgesehenen Maskenfelder ein. Anschließend klickt der Mandant auf den Button
                    „Weiter“.
                  </p>
                  <p>
                    <b>Kontaktdaten angeben:</b> <br />
                    Nun trägt der Mandant seine Adressdaten ein und klickt wiederum auf „Weiter“.
                  </p>
                  <p>
                    <b>Vollmacht und Auftrag erteilen:</b> <br />
                    Im letzten Schritt erteilt der Mandant der Kanzlei das Mandat und eine Vollmacht. Zugleich erhält
                    der Mandant eine Übersicht seiner Angaben. Sämtliche Angaben in der Auftragsübersicht können
                    jederzeit vor Anklicken des Buttons „Kostenpflichtig beauftragen“ auf der letzten Seite noch einmal
                    überprüft und korrigiert werden. In dem Feld „Digitale Unterschrift“ zeichnet der Mandant seinen
                    Namenszug. Durch Setzen eines Häkchens in die entsprechenden Kästchen bestätigt der Mandant die
                    Kenntnisnahme und das Einverständnis mit der Datenschutzerklärung sowie den Mandatsbedingungen.
                    Durch Anklicken der Schaltfläche "Kostenpflichtig beauftragen" erteilt der Mandant seinen Auftrag an
                    die Kanzlei und die Vollmacht.
                  </p>
                  <p>
                    <b>Nach Absenden des Auftrags:</b> <br />
                    <br />
                    Der Kanzlei schickt daraufhin dem Mandanten eine automatische Empfangsbestätigung per E-Mail zu, in
                    welcher die Vollmacht mit dem dahingehenden Vertretungsauftrag des Mandanten, die
                    Mandatsbedingungen, die abgeschlossene Vergütungsvereinbarung und die Widerrufsbelehrung nochmals
                    aufgeführt werden und die der Mandant über die Funktion „Drucken“ jeweils ausdrucken kann. Die
                    Annahme des Auftrags erfolgt separat nach Prüfung durch die Kanzlei.
                  </p>
                  <h2>§ 2 Vergütung und Zahlungsbedingungen</h2>
                  <p>
                    Die Vergütung der Kanzlei richtet sich nach der mit dem Mandanten abgeschlossenen
                    Vergütungsvereinbarung.
                  </p>
                  <h2>§ 3 Vertragstext</h2>
                  <p>
                    Der Vertragstext wird auf den internen Systemen der Kanzlei gespeichert. Die Auftragsdaten
                    (Vollmacht, ggf. Vergütungsvereinbarung) und die Mandatsbedingungen werden dem Mandanten per E-Mail
                    zugesendet. Nach Abschluss der Beauftragung sind die Auftragsdaten aus Sicherheitsgründen nicht mehr
                    über das Internet zugänglich.
                  </p>
                  <h2>§ 4 Schlussbestimmungen</h2>
                  <p>(1) Die Vertragssprache ist deutsch.</p>
                  <p>
                    (2) Auf Verträge zwischen der Kanzlei und den Mandanten findet das Recht der Bundesrepublik
                    Deutschland Anwendung. Diese Rechtswahl gilt bei Verbrauchern nur, soweit der durch zwingende
                    Bestimmungen des Rechts des Staates des gewöhnlichen Aufenthaltes des Verbrauchers gewährte Schutz
                    dem Mandanten nicht entzogen wird.
                  </p>
                  <p>
                    (3) Sofern es sich beim Mandanten um einen Kaufmann, eine juristische Person des öffentlichen Rechts
                    oder um ein öffentlich-rechtliches Sondervermögen handelt, ist Gerichtsstand für alle Streitigkeiten
                    aus Vertragsverhältnissen zwischen dem Mandanten und der Kanzlei der Sitz der Kanzlei. Dies gilt
                    auch, sofern der Mandant keinen allgemeinen Gerichtsstand in Deutschland oder der EU hat, oder sein
                    Wohnsitz oder sein gewöhnlicher Aufenthalt im Zeitpunkt der Klageerhebung nicht bekannt ist.
                  </p>
                </div>
              </DialogArticle>
            ),
          },
        ]}
        lockTarget="#storybook-root"
        name="dialog"
        position="top"
        size="m"
        onOverlayClick={() => {}}
        onClose={() => {}}
      />
    );
  },
};

export const DialogArticleWithoutCloseButton: Story = {
  name: "Dialog Article Without Close Button",
  args: {
    hasNoCloseButton: false,
  },
};

export const DialogArticleWithCloseButton: Story = {
  name: "Dialog Article With Close Button",
  args: {
    hasNoCloseButton: true,
  },
};

export default meta;
