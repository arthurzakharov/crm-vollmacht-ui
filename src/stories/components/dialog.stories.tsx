import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { Dialog, DialogPosition, DialogSize } from "../../components";
import Placeholder from "../helpers/placeholder";

type Story = StoryObj<typeof Dialog>;

const meta: Meta<typeof Dialog> = {
  title: "COMPONENTS/Dialog",
  component: Dialog,
  args: {
    dialogs: [
      {
        name: "dialog",
        element: (
          <div>
            <h2>Dialog (1)</h2>
            <Placeholder />
          </div>
        ),
      },
    ],
    lockTarget: "#root",
    name: "dialog",
    position: "top",
    size: "m",
    onOverlayClick: fn(),
    onClose: fn(),
  },
  render: (args) => {
    const [name, setName] = useState<string | undefined>(args.name);
    const [position, setPosition] = useState<DialogPosition | undefined>(args.position);
    const [size, setSize] = useState<DialogSize | undefined>(args.size);

    const onOverlayClick = () => {
      setPosition(undefined);
      setSize(undefined);
      args.onOverlayClick();
    };

    const onClose = () => {
      setName(undefined);
      args.onClose();
    };

    return (
      <Dialog {...args} name={name} position={position} size={size} onOverlayClick={onOverlayClick} onClose={onClose} />
    );
  },
};

export const PositionTop: Story = {
  name: "Position Top",
  args: {
    position: "top",
  },
};

export const PositionCenter: Story = {
  name: "Position Center",
  args: {
    position: "center",
  },
};

export const SizeS: Story = {
  name: "Size S",
  args: {
    size: "s",
  },
};

export const SizeM: Story = {
  name: "Size M",
  args: {
    size: "m",
  },
};

export const NameIsUndefined: Story = {
  name: "Name Is undefined",
  args: {
    name: undefined,
  },
};

export const DefaultValues: Story = {
  name: "Default Values",
  args: {
    lockTarget: undefined,
    name: undefined,
    position: undefined,
    size: undefined,
  },
};

export default meta;
