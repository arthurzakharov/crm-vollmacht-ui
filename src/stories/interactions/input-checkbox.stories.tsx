import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { expect, fireEvent, fn, userEvent, within } from "@storybook/test";
import { InputCheckbox } from "../../components";
import MaxWidthDecorator from "../decorators/max-width-decorator";

type Story = StoryObj<typeof InputCheckbox>;

const meta: Meta<typeof InputCheckbox> = {
  title: "INTERACTIONS/InputCheckbox",
  component: InputCheckbox,
  decorators: [MaxWidthDecorator],
  parameters: {
    MaxWidthDecorator: 600,
  },
  args: {
    value: false,
    name: "ageement",
    status: "neutral",
    disabled: false,
    onChange: fn(),
  },
  render: (args) => {
    const [value, setValue] = useState(args.value);
    const onChange = (value: boolean): void => {
      args.onChange(value);
      setValue(value);
    };
    return (
      <InputCheckbox {...args} value={value} onChange={onChange}>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam
        sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt
        dolor doloremque error <button type="button">explicabo</button> fuga fugit harum, illum ipsam, ipsum iste labore
        minima necessitatibus non numquam odio pariatur quam quas <a href="">quibusdam</a> quos reiciendis rem
        repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi
        corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur
        dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique
        sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita
        illo laudantium magnam natus non nulla porro qui, vero voluptas!
      </InputCheckbox>
    );
  },
};

export const Demo: Story = {
  name: "Demo",
  play: async ({ args, canvasElement, step }) => {
    const checkbox = within(canvasElement).getByRole("checkbox");
    const content = within(canvasElement).getByRole("checkbox").nextSibling as Element;
    const button = within(canvasElement).getByRole("button");
    const link = within(canvasElement).getByRole("link");
    await step("Click to ", async () => {
      await userEvent.click(content);
      await expect(args.onChange).toHaveBeenNthCalledWith(1, true);
    });
    await step("Use keyboard to select", async () => {
      await userEvent.tab();
      await expect(checkbox).toHaveFocus();
      await userEvent.keyboard("[Space]");
      await expect(args.onChange).toHaveBeenNthCalledWith(2, false);
      await expect(checkbox).toHaveFocus();
      await userEvent.tab();
      await expect(checkbox).not.toHaveFocus();
    });
    await step("Click on button and link in passed child", async () => {
      await userEvent.click(button);
      await expect(button).not.toHaveFocus();
      await expect(args.onChange).toHaveBeenCalledTimes(2);
      await userEvent.click(link);
      await expect(button).not.toHaveFocus();
      await expect(args.onChange).toHaveBeenCalledTimes(2);
    });
  },
};

export const Checked: Story = {
  name: "Checked",
  args: {
    value: true,
  },
};

export const NotChecked: Story = {
  name: "Not Checked",
  args: {
    value: false,
  },
};

export const StatusNeutral: Story = {
  name: "Status Neutral",
  args: {
    status: "neutral",
  },
};

export const StatusSuccess: Story = {
  name: "Status Success",
  args: {
    status: "success",
  },
};

export const StatusError: Story = {
  name: "Status Error",
  args: {
    status: "error",
  },
};

export const Disabled: Story = {
  name: "Disabled",
  args: {
    disabled: true,
  },
};

export const Enabled: Story = {
  name: "Enabled",
  args: {
    disabled: false,
  },
};

export const DefaultValues: Story = {
  name: "Default Values",
  args: {
    status: undefined,
    disabled: undefined,
  },
};

export default meta;
