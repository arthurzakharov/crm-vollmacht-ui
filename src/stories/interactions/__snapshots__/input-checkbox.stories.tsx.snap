// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INTERACTIONS/InputCheckbox Checked smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
           checked
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox DefaultValues smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox Demo play-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox Disabled smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input disabled
           type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked checkbox--disabled">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox Enabled smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox NotChecked smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox StatusError smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--error">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
    <div class="input-checkbox__icon">
      <div class="status-icon status-icon--error status-icon--m status-icon--active">
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox StatusNeutral smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--neutral">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`INTERACTIONS/InputCheckbox StatusSuccess smoke-test 1`] = `
<div style="max-width: 600px;">
  <div data-id="ageement"
       class="input-checkbox"
  >
    <input type="checkbox"
           tabindex="0"
           id="ageement"
           class="input-checkbox__element"
    >
    <div class="input-checkbox__content input-checkbox__content--success">
      <div class="input-checkbox__checkbox-wrap">
        <div class="checkbox checkbox--not-checked">
        </div>
      </div>
      <div class="input-checkbox__label">
        <label for="ageement"
               class="label label--s label--neutral"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolore eum incidunt magnam, quae quam quisquam sapiente unde. Accusantium adipisci blanditiis commodi consectetur consequuntur, cum cumque debitis deserunt dolor doloremque error
          <button type="button">
            explicabo
          </button>
          fuga fugit harum, illum ipsam, ipsum iste labore minima necessitatibus non numquam odio pariatur quam quas
          <a href>
            quibusdam
          </a>
          quos reiciendis rem repellendus reprehenderit repudiandae saepe sint sit ut veritatis voluptatibus voluptatum. Aliquid animi corporis ducimus earum, excepturi expedita ipsam maxime quaerat quas vitae. Aspernatur assumenda at consequuntur dolorum et, eveniet illum in, incidunt inventore itaque necessitatibus nihil odio, officia porro quas similique sit temporibus ut vero vitae! A adipisci, atque aut culpa cumque dicta dolorem ducimus ea error est expedita illo laudantium magnam natus non nulla porro qui, vero voluptas!
        </label>
      </div>
    </div>
  </div>
</div>
`;
